app/scripts/metamask-controller.js

```diff
@@ -516,6 +503,28 @@ export default class MetamaskController extends EventEmitter {
       ],
     });
 
+    this.queuedRequestController = new QueuedRequestController({
+      messenger: this.controllerMessenger.getRestricted({
+        name: 'QueuedRequestController',
+        allowedActions: [
+          'NetworkController:getState',
+          'NetworkController:setActiveNetwork',
+          'SelectedNetworkController:getNetworkClientIdForDomain',
+        ],
+        allowedEvents: ['SelectedNetworkController:stateChange'],
+      }),
+      shouldRequestSwitchNetwork: ({ method }) =>
+        methodsRequiringNetworkSwitch.includes(method),
+      canRequestSwitchNetworkWithoutApproval: ({ method }) =>
+        methodsThatCanSwitchNetworkWithoutApproval.includes(method),
+      clearPendingConfirmations,
+      showApprovalRequest: () => {
+        if (this.approvalController.getTotalApprovalCount() > 0) {
```

Comment:
This makes me wonder whether the confirmation window is getting triggered properly when the next batch is processed 🤔 e.g. after a batch with a smart transaction completes, the next queued batch might not trigger a popup.

Pre-existing problem, not a blocker here, but this might be a separate bug we need to fix.

Reply:
> after a batch with a smart transaction completes, the next queued batch might not trigger a popup.

Highly possible. I would assume there is some redundancy here where the ApprovalController may also be making sure the notification window is being shown? Unsure

Reply:
>I would assume there is some redundancy here where the ApprovalController may also be making sure the notification window is being shown? Unsure

Right, so there is. This should get triggered by the ApprovalController, we might be good here.